src/tests/ftest/control/log_entry.py

```diff
@@ -0,0 +1,186 @@
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+import contextlib
+import re
+
+from ClusterShell.NodeSet import NodeSet
+
+from apricot import TestWithServers
+from general_utils import get_journalctl, journalctl_time
+from run_utils import run_remote
+
+
+class ControlLogEntry(TestWithServers):
+    """Verify administrative log entries.
+
+    :avocado: recursive
+    """
+
+    def sleep(self, seconds):
+        """Sleep for some time.
+
+        Args:
+            seconds (int): number of seconds to sleep for
+        """
+        self.log.info('Sleeping for %s seconds', seconds)
+        time.sleep(seconds)
+
+    @contextlib.contextmanager
+    def verify_journalctl(self, expected_messages):
+        """Capture journalctl in a context and verify expected messages.
+
+        Args:
+            expected_messages (list): list of regular expressions to look for
+        """
+        t_before = journalctl_time()
+        yield
+        self.sleep(5)  # Give journalctl time to flush
+        t_after = journalctl_time()
+        self._verify_journalctl(t_before, t_after, expected_messages)
+
+    def _verify_journalctl(self, since, until, expected_messages):
+        """Verify journalctl contains one or more messages.
+
+        Args:
+            since (str): start time for journalctl
+            until (str): end time for journalctl
+            expected_messages (list): list of regular expressions to look for
+        """
+        self.log_step('Verify journalctl output between {} and {}'.format(since, until))
+
+        # Get the journalctl in the time-frame
+        results = get_journalctl(
+            hosts=self.hostlist_servers, since=since, until=until,
+            journalctl_type="daos_server")
+
+        # Convert the journalctl to a dict of hosts : output
+        journalctl_per_hosts = {}
+        for result in results:
+            journalctl_per_hosts[str(result['hosts'])] = result['data']
+
+        # For simplicity, look for each message in any host's journalctl
+        not_found = []
+        for message in expected_messages:
+            self.log.info('  Looking for %s', message)
+            if any(map(re.compile(message).search, journalctl_per_hosts.values())):
+                self.log.info('    FOUND')
+            else:
+                not_found.append(message)
+                self.log.info('    NOT FOUND')
+
+        # Print the journalctl in an easy-to-read format
+        for hosts, output in sorted(journalctl_per_hosts.items()):
+            self.log.debug('journalctl for %s:', hosts)
+            for line in output.splitlines():
+                self.log.debug('  %s', line)
+
+        # Fail if any message was not found
+        if not_found:
+            fail_msg = '{} messages not found in journalctl'.format(len(not_found))
+            self.log.error(fail_msg)
+            for message in not_found:
+                self.log.error('  %s', message)
+            self.fail(fail_msg)
```

Comment:
Just curious, why did you make the test fail instead of returning the result? I was thinking that not containing the expected message isn't so critical that we need to fail the test immediately.

Reply:
1. I think it's more difficult to figure out where a failure came from if, for example, we wait until after step 10 to declare that step 2 failed.
2. If some steps fail to log the correct messages, it's possible the system is in a weird stat that would cause subsequent steps to fail.
3. It's just easier to implement it this way :)